### PR TITLE
feat(DMS): add datasource smart connect tasks

### PR DESCRIPTION
**Pull request by 346010762**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  add datasource smart connect tasks
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/dms/ TESTARGS='-run TestAccDmsKafkaSmartConnectTasks_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dms/ -v -run TestAccDmsKafkaSmartConnectTasks_basic -timeout 360m -parallel 4
=== RUN   TestAccDmsKafkaSmartConnectTasks_basic
=== PAUSE TestAccDmsKafkaSmartConnectTasks_basic
=== CONT  TestAccDmsKafkaSmartConnectTasks_basic
--- PASS: TestAccDmsKafkaSmartConnectTasks_basic (1275.00s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dms       1275.040s
```

### CODE REVIEW

File: docs/data-sources/dms_kafka_smart_connect_tasks.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+---
+subcategory: "Distributed Message Service (DMS)"
+---
+
+# huaweicloud_dms_kafka_smart_connect_tasks
+
+Use this data source to get the list of DMS kafka smart connect task.
```
Comment (houpeng80) [inline docs/data-sources/dms_kafka_smart_connect_tasks.md]:
```suggestion
Use this data source to get the list of DMS kafka smart connect tasks.
```

File: docs/data-sources/dms_kafka_smart_connect_tasks.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+## Example Usage
+
+```hcl
+var "connector_id" {}
+
+data "huaweicloud_dms_kafka_smart_connect_tasks" "test" {
+  connector_id     = var.connector_id
+  task_id          = "60070a02-27c7-4eee-985d-95fc67f32d85"
```
Comment (houpeng80) [inline docs/data-sources/dms_kafka_smart_connect_tasks.md]: shoud use variable like connector_id

File: docs/data-sources/dms_kafka_smart_connect_tasks.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+* `status` - (Optional, String) Specifies the status of the smart connect task. Value options: **RUNNING**, **PAUSED**.
+
+## Attribute Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The data source ID.
+
+* `tasks` - The list of smart connect task.
```
Comment (houpeng80) [inline docs/data-sources/dms_kafka_smart_connect_tasks.md]:
```suggestion
* `tasks` - The list of smart connect tasks.
```

File: docs/data-sources/dms_kafka_smart_connect_tasks.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  The [tasks](#DMS_kafka_smart_connect_tasks) structure is documented below.
+
+<a name="DMS_kafka_smart_connect_tasks"></a>
+The `tasks` block supports:
+
+* `task_id` - Indicates the id of the smart connect task.
```
Comment (houpeng80) [inline docs/data-sources/dms_kafka_smart_connect_tasks.md]:
```suggestion
* `id` - Indicates the ID of the smart connect task.
```
and fix other places

File: docs/data-sources/dms_kafka_smart_connect_tasks.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+* `task_name` - Indicates the name of the smart connect task.
+
+* `destination_type` - Indicates the destination type of the smart connect task.
+
+* `created_at` - Indicates the creation time of the smart connect task.
+
+* `status` - Indicates the status of the smart connect task. The value can be: **RUNNING**, **PAUSED**.
+
+* `topics` - Indicates the topic names separated by a comma or the topic regular expression of the smart connect task.
```
Comment (houpeng80) [inline docs/data-sources/dms_kafka_smart_connect_tasks.md]:
```suggestion
* `topics` - Indicates the topic names separated by commas or the topic regular expression of the smart connect task.
```

File: docs/data-sources/dms_kafka_smart_connect_tasks.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String) Specifies the region in which to create the resource.
+  If omitted, the provider-level region will be used.
+
+* `connector_id` - (Required, String) Specifies the connector ID of the kafka instance.
+
+* `task_id` - (Optional, String) Specifies the id of the smart connect task.
```
Comment (houpeng80) [inline docs/data-sources/dms_kafka_smart_connect_tasks.md]:
```suggestion
* `task_id` - (Optional, String) Specifies the ID of the smart connect task.
```

File: docs/data-sources/dms_kafka_smart_connect_tasks.md (hunk 1 above)
Comment (346010762) [reply docs/data-sources/dms_kafka_smart_connect_tasks.md]: done

File: docs/data-sources/dms_kafka_smart_connect_tasks.md (hunk 2 above)
Comment (346010762) [reply docs/data-sources/dms_kafka_smart_connect_tasks.md]: done

File: docs/data-sources/dms_kafka_smart_connect_tasks.md (hunk 3 above)
Comment (346010762) [reply docs/data-sources/dms_kafka_smart_connect_tasks.md]: done

File: docs/data-sources/dms_kafka_smart_connect_tasks.md (hunk 4 above)
Comment (346010762) [reply docs/data-sources/dms_kafka_smart_connect_tasks.md]: done. change task_id as id in the list of tasks.

File: docs/data-sources/dms_kafka_smart_connect_tasks.md (hunk 5 above)
Comment (346010762) [reply docs/data-sources/dms_kafka_smart_connect_tasks.md]: done

File: docs/data-sources/dms_kafka_smart_connect_tasks.md (hunk 6 above)
Comment (346010762) [reply docs/data-sources/dms_kafka_smart_connect_tasks.md]: done

File: huaweicloud/services/dms/data_source_huaweicloud_dms_kafka_smart_connect_tasks.go [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Specifies the connector ID of the kafka instance.`,
+			},
+			"task_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `Specifies the ID of the smart connect task.`,
+			},
+			"task_name": {
```
Comment (ShiChangkuo) [inline huaweicloud/services/dms/data_source_huaweicloud_dms_kafka_smart_connect_tasks.go:39]: this parameter can be defined as `name` in the data source and tasksSchema

File: huaweicloud/services/dms/data_source_huaweicloud_dms_kafka_smart_connect_tasks.go [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+	dataSourceId, err := uuid.GenerateUUID()
+	if err != nil {
+		return diag.Errorf("unable to generate ID: %s", err)
+	}
+	d.SetId(dataSourceId)
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("connector_id", connectorID),
```
Comment (ShiChangkuo) [inline huaweicloud/services/dms/data_source_huaweicloud_dms_kafka_smart_connect_tasks.go]: it's unnecessary to set the required param in a data source.

Comment (346010762) [reply huaweicloud/services/dms/data_source_huaweicloud_dms_kafka_smart_connect_tasks.go]: deleted connector_id

**Pull request merged**